src/reducers/operations.ts

```diff
@@ -1,155 +1,344 @@
+    if (isOperationsAction(a)) {
+      if (a.type === 'Operations/Update') {
+        const action = a as IUpdateTransaction;
+        return {
+          ...state,
+          transactions: {
+            ...state.transactions,
+            [action.payload.id]: {
+              message: '',
+              ...state.transactions[action.payload.id],
+              ...action.payload.transaction
+            }
           }
-        }
+        };
       }
 
-      if (isDismissOperation(action)) {
-        const { hash } = action.payload;
-        return update(state, {$unset: [hash]})
+      if (a.type === 'Operations/Dismiss') {
+        const action = a as IDismissOperation;
+        const {
+          notifications: {[a.payload.id]: _, ...restNotifications},
+          transactions: {[a.payload.id]: __, ...restTransactions},
+        } = state;
+
+        return {
+          notifications: restNotifications,
+          transactions: restTransactions
+        };
       }
 
-      if (isShowOperation(action)) {
-        const hash = objectHash(action.payload);
-        return update(state, {
-          [hash]: {
-            $set: {
-              ...action.payload,
-              step: 1,
-              timestamp: +moment()
+      if (a.type === 'Operations/Show') {
+        const action = a as IShowNotification;
+        const { status, message, timestamp } = action.payload;
+        const id = action.payload.id;
+        return {
+          ...state,
+          notifications: {
+            ...state.notifications,
+            [id]: {
+              ...state.notifications[id],
+              status,
+              message,
+              timestamp
             }
           }
-        })
+        };
       }
+    }
 
-      return state;
-    };
+    return state;
+  }
 
 /**
- * A middleware that cancels a failed/successful action after timeout.
- * It respects AsyncAction.ttl if it's defined.
- * @param ttl default timeout before cancelling a failed action. zero means: do not cancel.
+ * Only persist pending transactions
  */
-export const actionCanceler =
-  (ttl: number = 5000): Middleware =>
-    (store) => (next) => (action: any) => {
-
-      if (isShowOperation(action) || (isAsyncAction(action) && action.sequence === AsyncActionSequence.Success)) {
-        const { operation } = action;
-        const timeout = (operation && typeof operation.ttl === 'number') ? operation.ttl : ttl;
-
-        const cancelAction =
-          isAsyncAction(action) ?
-            {
-              type: action.type,
-              meta: action.meta,
-              sequence: AsyncActionSequence.Cancel
-            } as IAsyncAction<any, any, any>
-          :
-            {
-              type: 'Operation/Dismiss',
-              payload: {
-                hash: objectHash(action.payload)
-              }
-            } as IDismissOperation;
+const filterPending = createTransform(
```

Comment:
this seems like a selector, you could use reselect and put in the selectors directory

Reply:
no not really, more I was suggesting that you check our reselect and see if that helps here. the nice thing about it is then it will cache the output of filterPending unless part of the state changes that is needed by the function